### ISSUE TITLE
react-countdown Helper function 'calcTimeDelta' returning wrong delta with different Timezone

### ISSUE THREAD

**Issue by AlexanderA7** [ndresx/react-countdown#111]
I am using the calcTimeDelta helper function and passing in the second optional argument the "now" parameter and passing in an alternative function that returns the date in milliseconds for MST. When the system clock is switched to another timezone, the countdown timer changes in a Chrome and Edge browser. It works properly in Firefox and does not change and maintains the countdown based on MST. I have added an image displaying the code I am using. The issues seems to be that "calcTimeDelta" does not use the Alternative function passed in via "now" parameter properly in Chrome and Edge. Change system clock to another timezone besides MST to test.

![ReactCountdownTimerIssue](https://user-images.githubusercontent.com/77363330/104382735-e458ba80-54eb-11eb-85ff-58f8808a1d3f.png)

**Comment by ndresx** [comment]
Hi @AlexanderA7, do you mean that there is a problem when the OS' timezone settings get switched by the user after the countdown was initiated, but is working fine otherwise, so when it's not getting changed during the countdown's lifetime? I have checked the code to see if something is not getting passed around correctly, but it should be fine from that perspective.

Could it eventually be the case, that `DateUtilities.MstNow()` is kind of caching the settings on initial load, but then not changed anymore? I'm not fully sure if I get the problem, but did you maybe try to replace the `now` implementation with `Date.now()`,  just to see if this is showing similar results to the ones you are experiencing right now?

**Comment by AlexanderA7** [comment]
Hi. During the countdown, if you change the OS timezone settings to a different timezone, the timer countdown adjusts based on that new timezone. It should not adjust at all but remain the same and countdown to the current time in MST regardless of what timezone the OS is in. The weird thing is that it works properly in Firefox browser, but not in Chrome and Edge.

**Comment by AlexanderA7** [comment]
In the included image. The left timer is from a Edge browser, the right timer is in Firefox. Both browsers are on the same computer set to Pacific Time. The correct timer is the one in the Firefox browser counting down to the MST time. They should both be displaying the 14 hours and not 15. The computer used to be in MST timezone setting. I changed that and the time in Edge changed but the timer in Firefox did not changed and kept counting down normally as expected.
![ReactCountdownTimerIssue-a](https://user-images.githubusercontent.com/77363330/104389176-12dc9280-54f8-11eb-88b8-06b882b64223.png)

**Comment by AlexanderA7** [comment]
Hi @Martin. Do you have any updates?

Alexander

**Comment by ndresx** [comment]
Hi, I have now played around with a couple of time variations on my machine, also with different online countdown services (e.g.: https://countingdownto.com/?c=3414636), and came to the conclusion, that this seems to be a browser-related problem and how each of them implemented their date and time handling when it comes to switching the OS' settings.

Although it seems to work the other way around for me, namely that Firefox considers these changes immediately, it takes some time for Chromium-based browsers to catch up. You could check this out again in this case https://codesandbox.io/s/great-feistel-f99cu (or any other online countdown services as said). 

At this point, I don't know if there is a solution to this particular problem because of the way the browsers work, and this countdown's reliance on JS' date-time functionality.

**Comment by AlexanderA7** [comment]
Hi. The issue I was experiencing turned out to be a lack of understanding of how the calcTimeDelta function works.
To make it work, we had to make the first option a  DateTime object (using the luxon library) that was in ISO format that depicted the time zone we wanted the time delta to be in. I have included the code that worked.

![ReactCountdownTimerSolution](https://user-images.githubusercontent.com/77363330/106327091-c94faf80-623a-11eb-9678-3cc9703ba00f.png)
